app/uk/gov/hmrc/hmrcemailrenderer/templates/twowaymessage/twoWayMessagingTemplates.scala

```diff
@@ -21,12 +21,16 @@ import uk.gov.hmrc.hmrcemailrenderer.templates.FromAddress
 import uk.gov.hmrc.hmrcemailrenderer.templates.ServiceIdentifier.TwoWayMessaging
 
 object twoWayMessagingTemplates {
+
+  private val getWaitTimeSubject: Map[String, String] => String =
```

Comment:
It's not obvious to me what this function does. Where does _ come from?

Reply:
Ah, so yeah that was a bit weird. But I guess it's to make it fairly versatile.

Part of the render request allows for a map called "Data" where you can easily add any old data onto it and deal with it later - because it's a map it doesn't require messing around with upstream/downstream domain objects. So this is just a nice way to agnostically grab stuff from it.

The subject actually wants a *function* which it'll use to get the subject, and later down the line that function will be applied to the data map of the RenderRequest.

 https://github.com/hmrc/hmrc-email-renderer/blob/d5a8e02446b8bbe4e6d08fb0daa558ae2da64231/app/uk/gov/hmrc/hmrcemailrenderer/domain/MessageTemplate.scala#L85

TBH, thinking about it. You've made me realise that I could just reuse the wait time.... i.e this should be...


`Map[String, String] => String = s"You will get a reply in _.getOrElse('waitTime', '7 days')"`

Reply:
Doesn't that defeat the object, in that you are hard-coding the subject text? Anyway I get the syntax now so I think it was fine how it was :)

Reply:
Yeahh, it's fine as is.